netplugin/netd.go

```diff
@@ -14,248 +14,299 @@ limitations under the License.
-			Network: opts.nwDriver,
+			Network: utils.OvsNameStr,
 			State:   stateStore,
 		},
 		Instance: core.InstanceInfo{
-			HostLabel:    opts.hostLabel,
-			CtrlIP:       opts.ctrlIP,
-			VtepIP:       opts.vtepIP,
-			UplinkIntf:   opts.vlanIntf,
-			DbURL:        opts.dbURL,
-			PluginMode:   opts.pluginMode,
-			VxlanUDPPort: opts.vxlanUDPPort,
+			HostLabel:    hostLabel,
+			CtrlIP:       controlIP,
+			VtepIP:       vtepIP,
+			UplinkIntf:   vlanUpLinks,
+			DbURL:        stateStoreURL,
+			PluginMode:   pluginMode,
+			VxlanUDPPort: vxlanPort,
+			// TODO: pass in network mode
+			FwdMode: forwardMode,
 		},
 	}
 
-	// Create a new agent
-	ag := agent.NewAgent(&pluginConfig)
-
-	// Process all current state
-	ag.ProcessCurrentState()
+	return pluginConfig, nil
+}
 
-	// post initialization processing
-	ag.PostInit()
+/*
+netplugin supported models:
+vxlan+routing, vlan+bridge/routing
+*/
 
-	// handle events
-	if err := ag.HandleEvents(); err != nil {
-		log.Infof("Netplugin exiting due to error: %v", err)
-		os.Exit(1)
+func main() {
+	hostname, _ := os.Hostname()
+	localIP, _ := cluster.GetLocalAddr()
+	app := cli.NewApp()
+	app.Version = "\n" + version.String()
+	app.Usage = "Contiv netplugin service"
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:   "log-level",
+			Value:  "INFO",
+			EnvVar: "CONTIV_NETPLUGIN_LOG_LEVEL",
+			Usage:  "set netplugin log level, options: [DEBUG, INFO, WARN, ERROR]",
+		},
+		cli.BoolFlag{
+			Name:   "use-json-log, json-log",
+			EnvVar: "CONTIV_NETPLUGIN_USE_JSON_LOG",
+			Usage:  "set netplugin log format to json",
+		},
+		cli.BoolFlag{
+			Name:   "use-syslog, syslog",
+			EnvVar: "CONTIV_NETPLUGIN_USE_SYSLOG",
+			Usage:  "set netplugin send log to syslog or not",
+		},
+		cli.StringFlag{
+			Name:   "syslog-url",
+			Value:  "udp://127.0.0.1:514",
+			EnvVar: "CONTIV_NETPLUGIN_SYSLOG_URL",
+			Usage:  "set netplugin syslog url in format protocol://ip:port",
+		},
+		cli.StringFlag{
+			Name:   "host, host-label",
+			Value:  hostname,
+			EnvVar: "CONTIV_NETPLUGIN_HOST",
+			Usage:  "set netplugin host to identify itself",
+		},
+		cli.StringFlag{
+			Name:   "mode, plugin-mode",
+			EnvVar: "CONTIV_NETPLUGIN_MODE",
+			Usage:  "set netplugin mode, options: [docker, kubernetes, swarm-mode]",
+		},
+		cli.StringFlag{
+			Name:   "vtep-ip",
+			Value:  localIP,
+			EnvVar: "CONTIV_NETPLUGIN_VTEP_IP",
+			Usage:  "set netplugin vtep ip for vxlan communication",
+		},
+		cli.StringFlag{
+			Name:   "ctrl-ip",
+			Value:  localIP,
+			EnvVar: "CONTIV_NETPLUGIN_CONTROL_IP",
+			Usage:  "set netplugin control ip for control plane communication",
+		},
+		cli.StringFlag{
+			Name:   "vlan-uplinks, vlan-if",
+			EnvVar: "CONTIV_NETPLUGIN_VLAN_UPLINKS",
+			Usage:  "a comma-delimited list of netplugin uplink interfaces",
+		},
+		cli.StringFlag{
+			Name:   "etcd-endpoints, etcd",
+			EnvVar: "CONTIV_NETPLUGIN_ETCD_ENDPOINTS",
+			Usage:  "a comma-delimited list of netplugin etcd endpoints",
```

Comment:
specify that consul-endpoints is ignored if this is set

Reply:
etcd endpoint will always take priority when it's set